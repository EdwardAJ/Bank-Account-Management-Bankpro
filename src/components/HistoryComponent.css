.wrapper {
    flex-wrap: wrap;
    display: flex;
    margin-top: 25px;
}