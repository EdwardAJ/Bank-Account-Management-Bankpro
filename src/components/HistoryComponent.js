import React from 'react';
import './HistoryComponent.css';

import axios from 'axios';
import { HISTORY_URL } from './../utils/endpoints.js';
import { HEADERS } from './../utils/header.js';
import getXMLResponse from './../utils/functions.js';
import { getCookie } from './../utils/cookie.js';

export default class HistoryComponent extends React.Component {
    constructor(props) {
        super(props);
        // Request and get response from backend.
        this.fetchHistory();
    }

    fetchHistory() {
        let requestBody =
        '<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:con="http://controllers.webservice.com/">' +
            '<soapenv:Header/>' +
            '<soapenv:Body>' +
                '<con:getHistory>' +
                    '<accountNumber>' + getCookie("auth") + '</accountNumber>' +
                '</con:getHistory>' +
            '</soapenv:Body>' +
        '</soapenv:Envelope>';

        axios.post(HISTORY_URL, requestBody, {
            headers: HEADERS
        }).then((response) => {
            if (response.status === 200) {
                let key = 0;
                for (let i = 0; i < getXMLResponse(response.data).length; i++) {
                    if (i % 4 === 0) {
                        key++
                    }
                    getXMLResponse(response.data)[i].innerHTML
                }
            }
        }).catch((error) => {
            console.log("Error: ", error);
        })

    }
}
