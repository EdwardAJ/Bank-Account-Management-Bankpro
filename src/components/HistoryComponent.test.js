import React from 'react';
import { shallow } from 'enzyme';
import HistoryComponent from './HistoryComponent';

describe("HistoryComponent", () => {
    it("should match this component", () => {
        const component = shallow(<HistoryComponent />);
        expect(component.getElements()).toMatchSnapshot();
    });
});

describe("HistoryComponent", () => {
    it("should render this component correctly", () => {
        const component = shallow(<HistoryComponent />);
        expect(component.find('.wrapper')).toHaveLength(1);
    });
});