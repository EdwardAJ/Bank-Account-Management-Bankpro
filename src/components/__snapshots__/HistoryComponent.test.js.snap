exports[`HistoryComponent should match this component 1`] = `
Array [
  <div
    className="wrapper">
    
</div>,
]
`;
